app/repositories/slick/implementations/SlickChatsRepository.scala

```diff
@@ -287,30 +580,37 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
     } yield row.addressId
   }
 
-  private[implementations] def insertEmailAddressIfNotExists(emailId: String, chatId: String, address: DBIO[String], participantType: String) /*: DBIO[String]*/ =
+  /**
+   * Method that inserts a new EmailAddressRow with a foreign key for an AddressRow
+   * @param emailId ID of the email
+   * @param chatId ID of the chat
+   * @param address  DBIOAction that returns the addressId (foreign key for the AddressesTable)
+   * @param participantType type of participant (from, to, bcc or cc)
+   * @return a DBIOAction with the number of inserted rows
+   */
+  private[implementations] def insertEmailAddress(emailId: String, chatId: String, address: DBIO[String], participantType: String) =
     for {
       addressId <- address
-      existing <- EmailAddressesTable.selectByEmailIdAddressAndType(emailId, addressId, participantType)
-        .result.headOption
-
-      row = existing
-        .getOrElse(EmailAddressRow(genUUID, emailId, chatId, addressId, participantType))
-
-      _ <- EmailAddressesTable.all.insertOrUpdate(row)
-    } yield ()
+      numberOfInsertedRows <- EmailAddressesTable.all += EmailAddressRow(genUUID, emailId, chatId, addressId, participantType)
+    } yield numberOfInsertedRows
 
+  /**
+   * Method that transforms an instance of the CreateChatDTO class into an instance of Chat
+   * @param chat instance of the class CreateChatDTO
+   * @return an instance of class Chat
+   */
   private[implementations] def fromCreateChatDTOtoChatDTO(chat: CreateChatDTO): Chat = {
     val email = chat.email
 
     Chat(
       chatId = chat.chatId.getOrElse(""),
       subject = chat.subject.getOrElse(""),
-      addresses = Set(email.from) ++ email.to.getOrElse(Set()) ++ email.bcc.getOrElse(Set()) ++ email.cc.getOrElse(Set()),
+      addresses = Set(email.from.orNull) ++ email.to.getOrElse(Set()) ++ email.bcc.getOrElse(Set()) ++ email.cc.getOrElse(Set()),
```

Comment:
orNull?

Reply:
It "returns the option's value if it is nonempty, or null if it is empty". What I want with that is a Set[String] that is either Set(email.from) or Set() (or Set(null)). I did this in some other way but IntelliJ suggested this correction with orNull

Reply:
null isn't an option in scala.

Reply:
email.from.toSet